src/span.js

```diff
@@ -0,0 +1,220 @@
+// @flow
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as constants from './constants.js';
+import * as thrift from './thrift.js';
+import Utils from './util.js';
+import SpanContext from './span_context.js';
+
+export default class Span {
+    _tracer: any;
+    _name: string;
+    _spanContext: SpanContext;
+    _start: number;
+    _duration: number;
+    _firstInProcess: boolean;
+    _isClient: boolean;
+    _peer: Endpoint;
+    _annotations: Array<Annotation>;
+    _binaryAnnotations: Array<BinaryAnnotation>;
+    _baggageHeaderCache: any;
+
+    constructor(tracer: any,
+                name: string,
+                spanContext: SpanContext,
+                start: number = Utils.getTimestampMicros(),
+                firstInProcess: boolean = false,
+                annotations: Array<Annotation> = [],
+                binaryAnnotations: Array<BinaryAnnotation> = []
+    ) {
+        this._tracer = tracer;
+        this._name = name;
+        this._spanContext = spanContext;
+        this._start = start;
+        this._firstInProcess = firstInProcess;
+
+        // logs
+        this._annotations = annotations;
+
+        // tags
+        this._binaryAnnotations = binaryAnnotations;
+    }
+
+    static _getBaggageHeaderCache() {
+        if (!this._baggageHeaderCache) {
+            // $FlowIgnore - flow doesn't seem to like how statics are declared
+            this._baggageHeaderCache = {};
```

Comment:
I don't see what makes this static - it looks just like any other member variable on `this`